### DISCUSSION TITLE
A Few Questions...

### DISCUSSION THREAD

**Discussion by RudraSen2** [facebook/docusaurus#8245] [Q&A]
I have a few queries:
1) The Back to top button only appears in the docs. How can I make it appear on every page?
2) I have my NavbarItem swizzled and ejected. I want to change the default button to the one in https://mui.com/joy-ui/react-button/ (Plain/Neutral/Small). How can I do it? I cannot understand the code after ejecting NavbarItem.
3) How can I set the footer.style to the current theme (light/dark)?

**Comment by slorber** [comment]
1) you can use the theme component `import BackToTopButton from '@theme/BackToTopButton';` in your custom pages. I guess we could add it to our blog too, and to our homepage as a demo?

2) I don't know what you mean by "default button". We don't have proper support for CSS-in-JS atm (https://github.com/facebook/docusaurus/issues/3236) and it's likely to create flash of unstyled content on page load due to MUI styles not being inlined in the HTML pages. Using a runtime CSS-in-JS lib is not really advised anymore for various reasons (https://dev.to/srmagura/why-were-breaking-up-wiht-css-in-js-4g9b). I'd recommend to not use MUI on a public-facing doc site in general which should remain performant and lightweight for visitors. For corporate React apps why not.

3) For now Footer seems to only have a static config-defined style, but I guess it would be easy to add support for a footer style that follows the current color mode of the site. You can probably swizzle it in the meantime and show us a demo of it. I don't even remember what the light footer looks like 😅

**Reply by RudraSen2** [reply]
for `BackToTopButton`, can I add it in `Root.js` so it gets applied on all pages?

**Reply by slorber** [reply]
No, because it will lead to a duplicate button on docs page (or you'd have to remove it first on docs page with swizzle to avoid that duplicate)

**Reply by complabs28** [reply]
I tried adding it on the Root.js file but it doesn't appear on other pages and neither it duplicates. And the CssVarsProvider has no effect on adding in root.

Code:

```js
import React from 'react';
import { getInitColorSchemeScript } from '@mui/joy/styles';
import { CssVarsProvider } from "@mui/joy/styles";
import BackToTopButton from '@theme/BackToTopButton';

// Default implementation, that you can customize
export default function Root({ children }) {
    return (
        <>
            {getInitColorSchemeScript()}
            <CssVarsProvider>
                {children}
            </CssVarsProvider>
        </>
    );
}
```

**Reply by RudraSen2** [reply]
I tried adding it on the Root.js file but it doesn't appear on other pages and neither it duplicates. And the CssVarsProvider has no effect on adding in root.

Code:

```js
import React from 'react';
import { getInitColorSchemeScript } from '@mui/joy/styles';
import { CssVarsProvider } from "@mui/joy/styles";
import BackToTopButton from '@theme/BackToTopButton';

// Default implementation, that you can customize
export default function Root({ children }) {
    return (
        <>
            {getInitColorSchemeScript()}
            <CssVarsProvider>
                {children}
            </CssVarsProvider>
        </>
    );
}
```

**Reply by slorber** [reply]
Please open a feature request for this. It probably make sense that Docusaurus add this button everywhere.

It would be quite easy to swizzle the button and disable it on a page/plugin basis, and if this becomes common we can add themeconfig options to make it possible without swizzle

**Reply by RudraSen2** [reply]
I have created the Feature Request long ago but forgot to put it here. https://docusaurus.io/feature-requests/p/add-back-to-top-button-on-every-page
